caom2utils/caom2utils/caom2blueprint.py

```diff
@@ -2587,70 +2578,60 @@ def _get_metrics(self, current):
+                self.logger.debug(f'Creating {label} Axis for {self.uri} from blueprint')
+
+            aug_error = self._two_param_constructor(
+                f'Chunk.{label}.axis.error.syser',
+                f'Chunk.{label}.axis.error.rnder',
+                index, _to_float, CoordError)
 
         aug_naxis = None
-        if aug_function is None:
-            aug_range = self._try_range_return(index, label)
-            if aug_axis is not None and aug_range is not None:
-                aug_naxis = CoordAxis1D(
-                    axis=aug_axis, error=aug_error, range=aug_range)
-                self.logger.debug(
-                    f'Creating range {label} CoordAxis1D for {self.uri} from '
-                    f'blueprint')
-        else:
-            if aug_axis is not None and aug_function is not None:
-                aug_naxis = CoordAxis1D(aug_axis, aug_error, None, None,
-                                        aug_function)
-                self.logger.debug(
-                    f'Creating function {label} CoordAxis1D for {self.uri} '
-                    f'from blueprint')
-        self.logger.debug(
-            f'End {label} naxis construction from blueprint.')
-        return aug_naxis
+        aug_range = self._try_range(index, label)
+        aug_naxis_index = None
+        if aug_axis is not None:
+            if aug_range is None:
+                if wcs is None or wcs.axis is None or wcs.axis.function is None:
+                    aug_ref_coord = self._two_param_constructor(
+                        f'Chunk.{label}.axis.function.refCoord.pix',
+                        f'Chunk.{label}.axis.function.refCoord.val',
+                        index, _to_float, RefCoord)
+                    aug_delta = _to_float(self._get_from_list(f'Chunk.{label}.axis.function.delta', index))
+                    aug_length = _to_int(self._get_from_list(f'Chunk.{label}.axis.function.naxis', index))
+                    aug_function = None
+                    if aug_length is not None and aug_delta is not None and aug_ref_coord is not None:
+                        aug_function = CoordFunction1D(aug_length, aug_delta, aug_ref_coord)
+                    aug_naxis = CoordAxis1D(aug_axis, aug_error, None, None, aug_function)
+                    if aug_function is not None:
+                        # if the WCS is described with a Function, cutouts can be supported, so specify an axis
+                        aug_naxis_index = _to_int(self._get_from_list(f'Chunk.{label}Axis', index))
+                    self.logger.debug(f'Creating function {label} CoordAxis1D for {self.uri} from blueprint')
+            else:
+                aug_naxis = CoordAxis1D(axis=aug_axis, error=aug_error, range=aug_range)
+                self.logger.debug(f'Creating range {label} CoordAxis1D for {self.uri} from blueprint')
+
+        self.logger.debug(f'End {label} axis construction from blueprint.')
+        return aug_naxis, aug_naxis_index
```

Comment:
WCS is something that I want to understand so I can make sense of this code.

Reply:
You and me both.